object-handlers.go

```diff
@@ -946,21 +946,40 @@ func (api objectAPIHandlers) CompleteMultipartUploadHandler(w http.ResponseWrite
 		completeParts = append(completeParts, part)
 	}
 	// Complete multipart upload.
+	// Send 200 OK
+	setCommonHeaders(w)
+	w.WriteHeader(http.StatusOK)
+
+	//Send whitespace character, once every 5secs, until CompleteMultipartUpload is done
+	doneCh := make(chan bool)
+	go func(doneCh <-chan bool) {
+		for {
+			select {
+			case <-time.After(5 * time.Second):
+				w.Write([]byte(" "))
```

Comment:
There is a race condition where w.Write() here can happen after writeErrorResponseNoHeader()

you can do this i think:
```
go func() {
    md5Sum, err = api.ObjectAPI.CompleteMultipartUpload(bucket, object, uploadID, completeParts)
    doneCh <- struct{}{}
}()

for {
select{
    case <-time.After(5 * time.Second): w.Write([]byte(" "))
    case <-doneCh:break
}
}
if err != nil {
    writeErrorResponseNoHeader()
    return
}
```